internal/services/storagecache/managed_lustre_file_system_resource.go

```diff
@@ -153,6 +161,35 @@ func (r ManagedLustreFileSystemResource) Arguments() map[string]*pluginsdk.Schem
 			},
 		},
 
+		"root_squash_setting": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"mode": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringInSlice(amlfilesystems.PossibleValuesForAmlFilesystemSquashMode(), false),
+					},
+					"no_squash_nid_list": {
```

Comment:
To avoid confusion on this property's type, we should drop `_list` from the name

```suggestion
					"no_squash_nids": {
```